docs/usage-notes.md

```diff
@@ -0,0 +1,52 @@
+This document shows several usage scenarios of SMACK that require special flags.
+
+## Bitwise Operations and Integer Casts
+If the program to verify contains bitwise operations or integer casts, then the flag `--bit-precise` may be required. The reason is that SMACK uses the SMT theory of integers to encode machine integers by default where bitwise operations are encoded using uninterpreted functions returning arbitrary values. Furthermore, special encoding is required to handle integer signedness casts, which is not also enabled automatically.
+
+The following program demonstrate the problems in the presence of bitwise operations.
+
+### Example
+```C
+#include "smack.h"
+
+int main () {
+  unsigned y = __VERIFIER_nondet_unsigned_int();
+  assume (y < 4U);
+  y >>= 2U;
+  assert (!y);
+}
+```
+This program should verify. However, if the we enable SMACK in its default mode, an error would be reported.
+
+```
+$ smack a.c
+SMACK program verifier version 1.9.0
+/home/shaobo/project/smack/install/share/smack/lib/smack.c(40,1): This assertion can fail
+Execution trace:
+    ...
+    a.c(4,16): Trace: Thread=1  (y = -669)
+    ...
+    a.c(6,5): Trace: Thread=1  (y = 17)
+    ...
+SMACK found an error.
+```
+
+Some steps in the error trace are omitted. As you can see, the concrete values of `y` in the error trace before and after the bitwise right shift operation do not follow its semantics because it is modeled as an uninterpreted function.
+
+For this case, we need the `--bit-precise` flag that, as its name indicates, enables bit-precise modeling of machine integers.
+
+```
+$ smack a.c --bit-precise
+SMACK program verifier version 1.9.0
+SMACK found no errors with unroll bound 1.
+```
+
+If the program that you would like to verify contains such patterns, please try this flag although it would cause SMACK to take much longer to perform the verification.
```

Comment:
If the program that you would like to verify contains such patterns, try enabling this flag. However, note that enabling bit-precise reasoning often degrades the performance of SMACK, and causes for it to take much longer to perform the verification.